src/main/capi/table_description-c.cpp

```diff
@@ -0,0 +1,78 @@
+#include "duckdb/main/capi/capi_internal.hpp"
+
+using duckdb::Connection;
+using duckdb::ErrorData;
+using duckdb::TableDescription;
+using duckdb::TableDescriptionWrapper;
+
+// FIXME: should the table name be its own struct for extensibility?
+// i.e duckdb_table_path
+// duckdb_table_path_set_schema(...)
+// duckdb_table_path_set_catalog(...)
```

Comment:
I like this. We can eventually use it to add another appender creation function allowing non-default schema appenders. Should we add it to this PR and change the parameters of `duckdb_table_description_create`?

Reply:
```c
//! The qualifier contains a path into the database.
//! Must be destroyed with `duckdb_qualifier_destroy`.
typedef struct _duckdb_qualifier {
	void *__qual;
} * duckdb_qualifier;
```

I really hope it doesn't need a destroy method, but since it will contain `std::string` variables, it probably needs it right?

Reply:
@Mytherin how do you feel about this?

Like Tania said, we are already suffering from this issue in the appender, possibly other places.
This way we wouldn't need to make a breaking change to add a `char *catalog` to `duckdb_appender_create` and can instead deprecate it, in favor of `duckdb_appender_create_with_qualifier` ?

Reply:
Let's just add a `duckdb_appender_create_ext` method that also takes the catalog, and leave the current method alone? I think adding the `duckdb_table_description` stuff seems overkill.

Reply:
The `duckdb_table_description` also serves other use cases such as getting the column count, `has_default`, `column_name`, etc. If a separate struct is overkill, we need to add this functionality to the appender. I.e., `duckdb_appender_column_has_default` and `duckdb_appender_column_name`.
cc @Mytherin

Reply:
By the way, there is already suggestion for `duckdb_appender_column_name` https://github.com/duckdb/duckdb/discussions/11737

Reply:
@Mytherin I think you're confusing `duckdb_qualifier` with `duckdb_table_description`
I agree `duckdb_qualifier` is overkill, I don't agree that `duckdb_table_description` is overkill

As Tania mentioned, the table description can be useful in other situations outside of the Appender, tying it to the Appender feels like the wrong concept

Reply:
I'm not against the `duckdb_table_description` - but I wonder if this is the right PR to introduce this?